airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/helpers/CatalogConverter.java

```diff
@@ -103,29 +104,57 @@ private static io.airbyte.protocol.models.AirbyteStream toConfiguredProtocol(fin
         .withNamespace(stream.getNamespace());
   }
 
-  public static io.airbyte.api.model.generated.AirbyteCatalog toApi(final io.airbyte.protocol.models.AirbyteCatalog catalog) {
+  public static io.airbyte.api.model.generated.AirbyteCatalog toApi(final io.airbyte.protocol.models.AirbyteCatalog catalog,
+                                                                    StandardSourceDefinition sourceDefinition) {
+    List<String> suggestedStreams = new ArrayList<>();
+    Boolean suggestingStreams;
+
+    // There are occasions in tests where we have not seeded the sourceDefinition fully. This is to
+    // prevent those tests from failing
+    if (sourceDefinition != null) {
+      suggestingStreams = sourceDefinition.getSuggestedStreams() != null;
+      if (suggestingStreams) {
+        suggestedStreams.addAll(sourceDefinition.getSuggestedStreams().getStreams());
+      }
+    } else {
+      suggestingStreams = false;
+    }
+
     return new io.airbyte.api.model.generated.AirbyteCatalog()
         .streams(catalog.getStreams()
             .stream()
             .map(CatalogConverter::toApi)
             .map(s -> new io.airbyte.api.model.generated.AirbyteStreamAndConfiguration()
                 .stream(s)
-                .config(generateDefaultConfiguration(s)))
+                .config(generateDefaultConfiguration(s, suggestingStreams, suggestedStreams)))
             .collect(Collectors.toList()));
   }
 
-  private static io.airbyte.api.model.generated.AirbyteStreamConfiguration generateDefaultConfiguration(final io.airbyte.api.model.generated.AirbyteStream stream) {
+  private static io.airbyte.api.model.generated.AirbyteStreamConfiguration generateDefaultConfiguration(final io.airbyte.api.model.generated.AirbyteStream stream,
+                                                                                                        Boolean suggestingStreams,
+                                                                                                        List<String> suggestedStreams) {
     final io.airbyte.api.model.generated.AirbyteStreamConfiguration result = new io.airbyte.api.model.generated.AirbyteStreamConfiguration()
         .aliasName(Names.toAlphanumericAndUnderscore(stream.getName()))
         .cursorField(stream.getDefaultCursorField())
         .destinationSyncMode(io.airbyte.api.model.generated.DestinationSyncMode.APPEND)
         .primaryKey(stream.getSourceDefinedPrimaryKey())
-        .selected(true);
+        .selected(!suggestingStreams)
+        .suggested(true);
+
+    if (suggestingStreams) {
```

Comment:
Nit: what about `result.setSelected(suggestingStreams && suggestedStreams.contains(stream.getName()))`? In general, I think `setValue(x)` is more readable than `if (x) setValue(true)`.

Reply:
🤦 noticed that it's setting different fields on the the result.

Not sure if that says anything about whether it could be made more readable, or that I should just read more carefully.